db/types/mathesar_char.py

```diff
@@ -0,0 +1,22 @@
+from sqlalchemy.dialects.postgresql import CHAR as SA_CHAR
+from sqlalchemy.types import TypeDecorator
+from sqlalchemy.ext.compiler import compiles
+
+from db.types.base import PostgresType
+
+
+class MATHESAR_CHAR(TypeDecorator):
+    impl = SA_CHAR
+
+    @classmethod
+    def __str__(cls):
+        return cls.__name__
+
+
+@compiles(MATHESAR_CHAR, 'postgresql')
+def _compile_mathesar_char(element, compiler, **kw):
+    unchanged_compiled_string = compiler.visit_VARCHAR(element, **kw)
+    unchanged_id = "VARCHAR"
+    changed_id = PostgresType.MATHESAR_CHAR.value.upper()
```

Comment:
This isn't correct in this case, since (as I've noted in a few spots) `"char"` is case-sensitive, and `"CHAR"` won't work.

Reply:
@mathemancer Are you implying to tell that I need to change line number 19 as  `unchanged_id = "char"`

Reply:
@mathemancer While compiling the MATHESAR_CHAR  the changed_compiled_string is returning VARCHAR rather than "char" which we have introduced via MATHESAR_CHAR class. As far as changed_id is concerned I have changed it to `PostgresType.MATHESAR_CHAR.value`